libcudacxx/include/cuda/std/source_location

```diff
@@ -0,0 +1,79 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX_SOURCE_LOCATION
+#define _LIBCUDACXX_SOURCE_LOCATION
```

Comment:
mild preference for giving our headers proper filename extensions.

Reply:
That ship has sailed with standard headers